extensions/wikia/AbPerfTesting/AbPerfTesting.setup.php

```diff
@@ -0,0 +1,124 @@
+	'author' => [
+		'[http://community.wikia.com/wiki/User:Macbre Maciej Brencz]'
+	],
+];
+
+// generic classes
+$wgAutoloadClasses[ 'Wikia\\AbPerfTesting\\Hooks'                     ] = __DIR__ . '/classes/Hooks.class.php';
+$wgAutoloadClasses[ 'Wikia\\AbPerfTesting\\UnknownCriterionException' ] = __DIR__ . '/classes/UnknownCriterionException.class.php';
+
+// experiments
+$wgAutoloadClasses[ 'Wikia\\AbPerfTesting\\Experiment'                 ] = __DIR__ . '/experiments/Experiment.class.php';
+$wgAutoloadClasses[ 'Wikia\\AbPerfTesting\\Experiments\\BackendDelay'  ] = __DIR__ . '/experiments/BackendDelay.class.php';
+$wgAutoloadClasses[ 'Wikia\\AbPerfTesting\\Experiments\\FrontendDelay' ] = __DIR__ . '/experiments/FrontendDelay.class.php';
+
+// criteria
+$wgAutoloadClasses[ 'Wikia\\AbPerfTesting\\Criterion'               ] = __DIR__ . '/criteria/Criterion.class.php';
+$wgAutoloadClasses[ 'Wikia\\AbPerfTesting\\Criteria\\OasisArticles' ] = __DIR__ . '/criteria/OasisArticles.class.php';
+$wgAutoloadClasses[ 'Wikia\\AbPerfTesting\\Criteria\\Traffic'       ] = __DIR__ . '/criteria/Traffic.class.php';
+$wgAutoloadClasses[ 'Wikia\\AbPerfTesting\\Criteria\\Wikis'         ] = __DIR__ . '/criteria/Wikis.class.php';
+
+// initialize the experiments when we have full page context available (skin, title, user, etc.)
+$wgHooks['AfterInitialize'][] = 'Wikia\\AbPerfTesting\\Hooks::onAfterInitialize';
+
+// experiments config goes below
+$wgABPerfTestingExperiments = [];
+
+/**
+ * Example entry:
+ *
+ * $wgABPerfTestingExperiments['an_experiment_42'] = [
+ *  # PHP class that will be created when the experiment is considered active
+ *	'handler' => 'Wikia\\AbPerfTesting\\Experiments\\AExperimentClass',
+ *
+ *  # parameters to be passed to the constructor as arguments
+ *	'params' => [
+ *		'foo' => 25,
+ *		'bar' => 42,
+ *	],
+ *
+ *  # set of criteria to which bucket given experiment should be assigned (all needs to be met)
+ *	'criteria' => [
+ *      # all wikis are split into 1000 buckets (modulo of city ID), pick one here
+ *		'wikis' => 1,
+ *      # all clients are split into 1000 buckets (modulo of beacon_id md5 hash), pick one here
+ * 		'traffic' => 1,
+ *      # run the test on Oasis and content namespaces only
+ * 		'oasisArticles' => true
+ *	]
+ *];
+ *
+ * Devbox debugging of traffic criterion:
+ *
+ * 1. Set the beacon cookie: document.cookie = 'wikia_beacon_id=3j-YqSr9BQ'
+ * 2. Set the "traffic" criterion to '158'
+**/
+
+$wgABPerfTestingExperiments['backend_delay_0'] = [
+	'handler' => 'Wikia\\AbPerfTesting\\Experiments\\BackendDelay',
+	'params' => [
+		'delay' => 0, // control group to compare other three groups against
+	],
+	'criteria' => [
+		'oasisArticles' => true,
+		'traffic' => 0,
+	]
+];
+
+$wgABPerfTestingExperiments['backend_delay_1'] = [
+	'handler' => 'Wikia\\AbPerfTesting\\Experiments\\BackendDelay',
+	'params' => [
+		'delay' => 100,
+	],
+	'criteria' => [
+		'oasisArticles' => true,
+		'traffic' => 1,
```

Comment:
how about support for ranges here, eg. `[ 1, 100 ]` or `'1-10'`? As we spoke we'll probably want to have more than 0.1% sampling over the traffic in a given bucket?